base/REPLCompletions.jl

```diff
@@ -133,12 +133,12 @@ function complete_path(path::AbstractString, pos; use_envpath=false)
         return UTF8String[], 0:-1, false
     end
 
-    matches = UTF8String[]
+    matches = Dict{UTF8String,Bool}()
```

Comment:
Yes, I wasn't aware of it being internally a `dict::Dict{T,Void}` until now. The documentation is really lacking there… it's not clear that it internally uses a hash map (or has O(1) complexity for access, which is the important part here).

I will probably amend this along with cleaning up the tests or so.